### DISCUSSION TITLE
Display config for display area on screen

### DISCUSSION THREAD

**Discussion by willemrosenthal** [OnionUI/Onion#1413] [Q&A]
Is there a cofig file or a way to alter the 'safe' area on the screen that the os will display on? 

I'm building custom hardware and need to use a screen where not 100% of the display will be visible. I've done this with various raspberry pi OSs but not sure how to do it with Onion OS

**Comment by XK9274** [comment]
Are you creating a new OS or frontend? Cuz that could be done at software level with SDL no?

The panel is initialised by: https://wx.comake.online/doc/d8clf27cnes2-SSD20X/customer/development/mi/en/mi_panel.html
The display is initialised by: https://wx.comake.online/doc/d8clf27cnes2-SSD20X/customer/development/mi/en/mi_disp.html

I'm pretty sure MainUI is hardcoded to 640x480, or SDL is. The panel most definitely is:


```
PanelName:K101_IM2BVL

--------------------------------------- PANEL Dev0 Info -------------------------------------
  LVDS_POL        LVDS_CH      LINK_TYPE        TI_MODE
         0              0       MIPI_DSI              1
    SW_ODD        SW_EVEN      SW_ODD_RB     SW_EVEN_RB
         0              0              0              0
   H_Total        V_Total          Width         Height        H_Start        V_Start
       780            515            640            480             42             18
       hbp           hspw            hfp            vbp           vspw            vfp
        24             20             96             13              5             17
      DClk      FrameRate       INV_DCLK         INV_DE       InvHSync       InvVSync
        26              0              0              0              1              1
SSC_Enable       SSC_Step       SSC_Span         TI_BIT         Format        chnswap
   disable              0              0           8BIT           8BIT       (2,4,3,1,0)
--------------------------------MIPI CONFIG-------------------------------
   HsTrail         HsPrpr         HsZero      ClkHsPrpr      ClkHsExit       ClkTrail
         3              2              5              2              4              4
   ClkZero      ClkHsPost       DaHsExit        ContDet            Lpx          TaGet
        10             10              4              0             16             26
    TaSure           TaGo        Hactive            Hpw            Hbp            Hfp
        22             50            640            120            120            120
   Vactive            Vpw            Vbp            Vfp           Bllp            Fps
       480              5             13             17              0             60
  lanemode         format       ctrlmode  SyncCalibrate     VirHsyncSt    VirHsyncEnd       VsyncRef    DataClkSkew
     2lane         rgb888     sync_pulse              1            431            441          32978              9
   PolChn0        PolChn1        PolChn2        PolChn3        PolChn4
         0              0              0              0              0

```

And MI_Disp is also pinned to this res:

```
----------------------------------------BindPeerInputPortList---------------------------------------
  ChnId  PassId  PortId  Enable  bind_module_id  bind_module_name  bind_ChnId  bind_PortId bind_Type   bind_Param  enable
============================== Private DISP0 Info ============================
 DevStatus         IrqNum         IrqCnt        BgColor
         1             57         180709         800080
 Interface      DevTiming      CscMatrix           Luma       Contrast            Hue     Saturation      Sharpness
       LCD        UNKNOWN              3             44             50             65             50              0
--------------------------------- Layer Info ---------------------------------
   LayerId    BindedDevID     LayerWidth    LayerHeight
         0              0              0              0
   LayerId   LayDispWidth  LayDispHeight     Toleration     rotatemode
         0              0              0              0           NONE
--------------------------------- Layer0 InputPort Info ---------------------------------
    PortId    enable CurStatus     src_w     src_h    crop_x    crop_y    crop_w    crop_h    show_x    show_y    show_w    show_h
         0         1         0       640       480         0         0         0         0         0         0       640       480

```

Just to add aswell, the framebuffer is also set at 640x480:

```
/ # fbset

mode "640x480-75"
        # D: 31.500 MHz, H: 37.500 kHz, V: 75.000 Hz
        geometry 640 480 640 960 32
        timings 31746 120 16 16 1 64 3
        accel false
        rgba 8/16,8/8,8/0,8/24
endmode
```

This isn't an Onion OS thing, it's a Sigmastar thing (the SoC mfc)

**Comment by willemrosenthal** [comment]
No intentions to create a new OS / Frontend. I'm honestly new to Onion OS, just working on a little modding project and hoping I can use the OS as it is– just with a different screen display area. (Messing with a Miyoo Mini that I'd be installing Onion on)

So are these files you sent just editable configs? Sorry, but I'm totally new to this type of work. Is it as simple as updating the pixel dimensions on these, or is there a lot more involved?

Where would I find these files in the file structure? And sorry if this is a dumb question, but is this something I'd edit before or after install?

**Reply by XK9274** [reply]
> No intentions to create a new OS / Frontend. I'm honestly new to Onion OS, just working on a little modding project and hoping I can use the OS as it is– just with a different screen display area. (Messing with a Miyoo Mini + that I'd be installing Onion on)
> 
> So are these files you sent just editable configs? Sorry, but I'm totally new to this type of work. Is it as simple as updating the pixel dimensions on these, or is there a lot more involved?

Ah ok I see! It's a great little device to tinker with!

mi_disp/mi_panel are modules provided by the SOC that control the initialisation of the display and panel to set dimensions.

So Onion sits on top of MainUI. MainUI is Miyoos (closed source) frontend, which uses SDL but also initialises the panel via some modules provided by the SoC MFC.

There's also some uboot init of the screen but that all happens before the kernel loads.. anyway...

All of this is done to 640x480 (different on v4 1027fw)

You could try to push geometry changes into `fbset` but you'll likely get a broken image or flickering/slideshow & this will get reset quite often.

They're not config files no, they're outputs from modules already initialised by Miyoos kernel/MainUI & a binary called `l`. You can push changes into these modules with echos but in all the testing I've done it's never resulted in a resized MainUI (it ends up cropping the viewport instead of rescaling it & "blank" areas have a strange repeating vertical pattern)

But to put it honestly, I don't know of any easy way to scale the display output of MainUI and SDL or crop/relocate it. (currently).. but that's not to say one doesn't exist 👍

**Reply by willemrosenthal** [reply]
Ah, too bad. Really appreciate the feedback regardless!
